include/libopencm3/stm32/g0/adc.h

```diff
@@ -114,117 +135,147 @@
 @{*/
 
 /** ADVREGEN: Voltage regulator enable bit */
-#define ADC_CR_ADVREGEN				(1 << 28)
+#define ADC_CR_ADVREGEN		(1 << 28)
 
 /**@}*/
 
 /** @addtogroup adc_cfgr1
 @{*/
 
 /** ALIGN: Data alignment */
-#define ADC_CFGR1_ALIGN				(1 << 5)
+#define ADC_CFGR1_ALIGN			(1 << 5)
 
 /* EXTSEL[2:0]: External trigger selection for regular group */
-#define ADC_CFGR1_EXTSEL_SHIFT			6
-#define ADC_CFGR1_EXTSEL			(0x7 << ADC_CFGR1_EXTSEL_SHIFT)
-#define ADC_CFGR1_EXTSEL_VAL(x)			((x) << ADC_CFGR1_EXTSEL_SHIFT)
+#define ADC_CFGR1_EXTSEL_SHIFT		6
+#define ADC_CFGR1_EXTSEL_MASK		(0x7 << ADC_CFGR1_EXTSEL_SHIFT)
+#define ADC_CFGR1_EXTSEL_VAL(x)		((x) << ADC_CFGR1_EXTSEL_SHIFT)
+/** @defgroup adc_cfgr1_extsel ADC external trigger selection values
+ *@{*/
+#define ADC_CFGR1_EXTSEL_TIM1_TRGO2	0x0
+#define ADC_CFGR1_EXTSEL_TIM1_CC4	0x1
+#define ADC_CFGR1_EXTSEL_TIM2_TRGO	0x2
+#define ADC_CFGR1_EXTSEL_TIM3_TRGO	0x3
+#define ADC_CFGR1_EXTSEL_TIM15_TRGO	0x4
+/**@}*/
 
 /** CHSELRMOD: Mode Selection of the ADC_CHSELR register */
-#define ADC_CFGR1_CHSELRMOD			(1 << 21)
+#define ADC_CFGR1_CHSELRMOD		(1 << 21)
 
 /**@}*/
 
 /** @addtogroup adc_cfgr2
 @{*/
 
-#define ADC_CFGR2_CKMODE_SHIFT		(30)
-#define ADC_CFGR2_CKMODE_MASK		(0x3)
+#define ADC_CFGR2_CKMODE_SHIFT		30
+#define ADC_CFGR2_CKMODE_MASK		(0x3 << ADC_CFGR2_CKMODE_SHIFT)
+#define ADC_CFGR2_CKMODE_VAL(x)		((x) << ADC_CFGR2_CKMODE_SHIFT)
 /** @defgroup adc_cfgr2_ckmode ADC Clock mode
  *@{*/
-#define ADC_CFGR2_CKMODE_ADCCLK		(0x0)
-#define ADC_CFGR2_CKMODE_PCLK_DIV2	(0x1)
-#define ADC_CFGR2_CKMODE_PCLK_DIV4	(0x2)
-#define ADC_CFGR2_CKMODE_PCLK		(0x3)
+#define ADC_CFGR2_CKMODE_ADCCLK		0x0
+#define ADC_CFGR2_CKMODE_PCLK_DIV2	0x1
+#define ADC_CFGR2_CKMODE_PCLK_DIV4	0x2
+#define ADC_CFGR2_CKMODE_PCLK		0x3
 /**@}*/
 
 /** LFTRIG: Low Frequency Trigger Mode enable bit */
-#define ADC_CFGR2_LFTRIG			(1 << 29)
+#define ADC_CFGR2_LFTRIG		(1 << 29)
 
 /** TOVS: Triggered Oversampling */
-#define ADC_CFGR2_TOVS				(1 << 9)
+#define ADC_CFGR2_TOVS			(1 << 9)
 
-#define ADC_CFGR2_OVSS_SHIFT		(5)
-#define ADC_CFGR2_OVSS_MASK			(0xf)
-/** @defgroup adc_cfgr2_ovss ADC Oversampling shift
- *@{*/
-#define ADC_CFGR2_OVSS_BITS(bits)	(bits)
-/**@}*/
+#define ADC_CFGR2_OVSS_SHIFT		5
+#define ADC_CFGR2_OVSS_MASK		(0xf << ADC_CFGR2_OVSS_SHIFT)
+#define ADC_CFGR2_OVSS_VAL(x)		((x) << ADC_CFGR2_OVSS_SHIFT)
 
-#define ADC_CFGR2_OVSR_SHIFT		(2)
-#define ADC_CFGR2_OVSR_MASK			(0x7)
+#define ADC_CFGR2_OVSR_SHIFT		2
+#define ADC_CFGR2_OVSR_MASK		(0x7 << ADC_CFGR2_OVSR_SHIFT)
+#define ADC_CFGR2_OVSR_VAL(x)		((x) << ADC_CFGR2_OVSR_SHIFT)
```

Comment:
you've got this a few times.  You've made some changes in the right direction, towards unshifted defines, as requested/desired, but then you have a few that go backwards.  I've picked out about all I can out of this change for now.  I don't think I can take any more out of this current changeset.

Reply:
Wait, wait.. The values are all unshifted as you requested. `_MASK` and `_VAL` are shifted, yes. As they are everywhere else.

Reply:
`*_VAL` are shifted everywhere, masks are mixed. So what do you want me to do - unshift masks, leave `_VAL` shifted?

Reply:
I actually want _VAL foisted onto the user to be gone.  some of these functyions are just layered macros for no reason.  the _user_ facing api should be unshifted, so they can use natural numbers in their own applications.  the library shouldhandle all shifts/masks internally where required.